### PR TITLE
Harden sync client contract coverage

### PR DESCRIPTION
**Pull request by sachaservan**
## Summary
- add CI-covered unit tests for sync request boundaries, authentication, response validation, serialization, cancellation, and revision behavior
- reject pre-aborted sync operations before authentication or network work begins
- remove the unusable live sync suite and workflow because Clerk session JWTs cannot be maintained as CI secrets
- make upload retry timing deterministic to prevent random zero-delay failures

## Testing
- `npm run test:unit -- --run` (152 files, 1,742 tests)
- `npm run lint`
- upload coalescer suite passed 20 consecutive runs

### CODE REVIEW

File: src/services/sync-enclave/sync-enclave-client.ts [hunk 1]
```diff
@@ -341,6 +341,7 @@ async function runBoundedOperation<T>(
   activeOperationControllers.add(controller)
   const timer = setTimeout(() => controller.abort(timeoutError()), timeoutMs)
   try {
+    if (controller.signal.aborted) throw controller.signal.reason
```
[the remark this answers is not in the document]
Comment (sachaservan) [reply src/services/sync-enclave/sync-enclave-client.ts:344]: Keeping this guard: `operation(controller.signal)` is evaluated before `settleWithSignal` can inspect the signal, and request setup can synchronously invoke the token getter while constructing that promise. The guard is what makes pre-aborted requests avoid authentication work, as asserted by the test.

File: tests/integration/sync-enclave/smoke.test.ts [hunk 2]
```diff
@@ -66,29 +107,81 @@ describe.skipIf(!enabled)('sync-enclave live smoke', () => {
...
-    // Best-effort cleanup. The enclave is the source of truth, so a
-    // leftover key wastes nothing but it does muddy subsequent runs.
+    const cleanupFailures: unknown[] = []
+    for (const id of attachmentIds) {
+      await collectCleanupFailure(
+        () => attachmentDelete({ id }),
+        cleanupFailures,
+      )
+    }
+    if (chatEtag && !chatDeleted) {
```
[the remark this answers is not in the document]
Comment (sachaservan) [reply tests/integration/sync-enclave/smoke.test.ts]: Leaving these two explicit cleanup branches. They are short, differ in meaningful row state, and a parameterized helper would add indirection without reducing error-handling complexity.

File: tests/integration/sync-enclave/smoke.test.ts [hunk 3]
```diff
@@ -137,13 +224,191 @@ describe.skipIf(!enabled)('sync-enclave live smoke', () => {
...
+      idempotencyKey: newIdempotencyKey(),
+    }
+    const stored = await attachmentPut(putRequest)
+    attachmentIds.add(stored.id)
+
+    await expect(attachmentPut(putRequest)).resolves.toEqual(stored)
+    const alteredPlaintext = new Uint8Array([1, 2, 3])
+    const altered = await attachmentPut({
+      ...putRequest,
+      plaintext: alteredPlaintext,
```
[the remark this answers is not in the document]
Comment (sachaservan) [reply tests/integration/sync-enclave/smoke.test.ts]: Fixed in a7516db2 by minting a fresh idempotency key for the altered attachment write.

File: tests/integration/sync-enclave/smoke.test.ts [hunk 4]
```diff
@@ -137,13 +224,191 @@ describe.skipIf(!enabled)('sync-enclave live smoke', () => {
...
+      direction: 'desc',
+      limit: 100,
+    })
+    expect(statusAfterDelete.deletes.some((item) => item.id === chatId)).toBe(
+      true,
+    )
+    const restoredPayload = new TextEncoder().encode(
+      JSON.stringify({ id: chatId, title: 'restored', runId }),
+    )
+    const restored = await push({
```
[the remark this answers is not in the document]
Comment (sachaservan) [reply tests/integration/sync-enclave/smoke.test.ts]: Fixed in a7516db2 by setting `metadata.restoreDeleted` on the tombstone restore write.
